### DISCUSSION TITLE
[Feature] I made a Twitch-like command sytem. What do you think about ?

### DISCUSSION THREAD

**Discussion by SkYNewZ** [owncast/owncast#473] [General Owncast discussion]
#### Features
- Define commands in a `.yaml` file
- Write placeholders (for dynamic commands - like `!uptime` for example)
- Use placeholders in the `commands.yml` file
- Support answer emojis in commands with https://github.com/kyokomi/emoji

It is work in progress but it is working.

https://github.com/SkYNewZ/owncast-commands

**Comment by gabek** [comment]
How cool, nice job!

I love the idea of building external bots to do stuff like this, I can't wait to see what you continue to add. If I think of any commands myself I'll totally send them your way!

I just wanted to give a heads up, though.  At the moment, like you know, anybody can connect to the websocket and do whatever you want.  And that's cool for stuff like this, but we do need to figure out some way to protect it a little from just _anyone_ connecting and spamming or whatever.  We haven't figured out what the process would be for that yet, but I totally don't want to lock out people from building their own tools on top of the chat.  https://github.com/owncast/owncast/issues/321

**Comment by gabek** [comment]
Hey @SkYNewZ!  Just wanted to point you to the work going on over at #556, hopefully it should facilitate making things like your command system even easier.  Provide any thoughts or feedback!
